Mathlib/Topology/Semicontinuous.lean

```diff
@@ -154,6 +166,12 @@ theorem LowerSemicontinuousOn.mono (h : LowerSemicontinuousOn f s) (hst : t ⊆
 theorem lowerSemicontinuousOn_univ_iff : LowerSemicontinuousOn f univ ↔ LowerSemicontinuous f := by
   simp [LowerSemicontinuousOn, LowerSemicontinuous, lowerSemicontinuousWithinAt_univ_iff]
 
+theorem lowerSemicontinuousOn_iff_restrict :
+    LowerSemicontinuousOn f s ↔ LowerSemicontinuous (s.restrict f) := by
```

Comment:
If we flip the LHS and RHS then this can be a useful `simp` lemma:
```suggestion
@[simp] theorem lowerSemicontinuous_restrict_iff :
    LowerSemicontinuous (s.restrict f) ↔ LowerSemicontinuousOn f s := by
```